code_generator/GdkCodeGenerator/Templates/UnityComponentConversionGenerator.tt

```diff
@@ -287,15 +310,91 @@ namespace <#= qualifiedNamespace #>
                 );
             }
 
-<# foreach (var command in commandDetailsList) { #>
-            private void On<#= command.CommandName #>Request(CommandRequestOp op)
+<# foreach (var commandDetails in commandDetailsList) {
+    var wrappedCommandRequestType = $"{qualifiedNamespace}.{componentDetails.ComponentName}.{commandDetails.CommandName}.ReceivedRequest";
+    var commandRequestBufferType = $"{qualifiedNamespace}.{componentDetails.ComponentName}.CommandRequests.{commandDetails.CommandName}";
+
+    var wrappedCommandResponseType = $"{qualifiedNamespace}.{componentDetails.ComponentName}.{commandDetails.CommandName}.ReceivedResponse";
+    var commandResponseBufferType = $"{qualifiedNamespace}.{componentDetails.ComponentName}.CommandResponses.{commandDetails.CommandName}";
+
+    var commandStorage = $"{commandDetails.CamelCaseName}Storage";
+#>
+            private void On<#= commandDetails.CommandName #>Request(CommandRequestOp op)
             {
-                // TODO: Implement commands
+                if (!IsValidEntityId(op.EntityId, "CommandRequestOp", out var entity))
+                {
+                    return;
+                }
+
+                var deserializedRequest = <#= commandDetails.RequestType #>.Serialization.Deserialize(op.Request.SchemaData.Value.GetObject());
+                if (entityManager.HasComponent<<#= commandRequestBufferType #>>(entity))
+                {
+                    var data = entityManager.GetComponentData<<#= commandRequestBufferType #>>(entity);
+                    data.Requests.Add(new <#= wrappedCommandRequestType #>(op.RequestId.Id,
+                        op.CallerWorkerId,
+                        op.CallerAttributeSet,
+                        deserializedRequest));
+                }
+                else
+                {
+                    var data = new <#= commandRequestBufferType #>();
```

Comment:
This bit does a lot of extra calls
The handle can be provided inside the `new` bit
The List can be cached to add the element to, to avoid a lookup
`var list = data.Requests = new List<<#= wrappedCommandRequestType #>>();`